hbase-common/src/test/java/org/apache/hadoop/hbase/io/crypto/tls/TestX509Util.java

```diff
@@ -142,6 +142,38 @@ public void cleanUp() {
     Security.setProperty("com.sun.security.enableCRLDP", Boolean.FALSE.toString());
   }
 
+  @Test
+  public void testCreateSSLContextWithClientAuthDefault() throws Exception {
```

Comment:
I add tests for ClientAuth passthrough here, but it's not really possible (as far as i can tell) to access the TrustManager of the underlying SslContext/SSLEngine. So I can't easily add tests here for the TrustManager hostname verification arg passthrough. So instead we have the comprehensive end-to-end tests in TestMutualTlsClientSide/ServerSide